server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -188,11 +254,196 @@ public Result updateVisibility(Request request, long programId, long blockDefini
+              request, programDefinition, blockDefinition, questionDefinitions));
+    } catch (ProgramNotFoundException | ProgramBlockDefinitionNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result configureNewEligibilityPredicate(
+      Request request, long programId, long blockDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    DynamicForm form = formFactory.form().bindFromRequest(request);
+
+    try {
+      ProgramDefinition programDefinition = programService.getProgramDefinition(programId);
+      BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockDefinitionId);
+
+      return ok(
+          predicatesConfigureView.renderEligibility(
+              request,
+              programDefinition,
+              blockDefinition,
+              getQuestionDefinitionsForForm(programDefinition, blockDefinition, form)));
+    } catch (ProgramNotFoundException | ProgramBlockDefinitionNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result configureExistingEligibilityPredicate(
+      Request request, long programId, long blockDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    try {
+      ProgramDefinition programDefinition = programService.getProgramDefinition(programId);
+      BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockDefinitionId);
+
+      ImmutableList<Long> questionIds =
+          blockDefinition
+              .eligibilityDefinition()
+              .orElseThrow(
+                  () ->
+                      new RuntimeException(
+                          String.format(
+                              "Block %d has no eligibility definition", blockDefinition.id())))
+              .predicate()
+              .getQuestions();
+
+      ImmutableList<QuestionDefinition> questionDefinitions =
+          getQuestionDefinitions(
+              programDefinition,
+              blockDefinition,
+              (QuestionDefinition questionDefinition) ->
+                  questionIds.contains(questionDefinition.getId()));
+
+      return ok(
+          predicatesConfigureView.renderEligibility(
+              request, programDefinition, blockDefinition, questionDefinitions));
+    } catch (ProgramNotFoundException | ProgramBlockDefinitionNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ImmutableList<QuestionDefinition> getQuestionDefinitionsForForm(
+      ProgramDefinition programDefinition, BlockDefinition blockDefinition, DynamicForm form) {
+    return getQuestionDefinitions(
+        programDefinition,
+        blockDefinition,
+        (QuestionDefinition questionDefinition) ->
+            form.rawData().containsKey(String.format("question-%d", questionDefinition.getId())));
```

Comment:
We should ensure that the form only contains questions that are valid in the block (eg staleness detection), this will otherwise ignore them and keep none or a partial set of them.

Reply:
That's handled in `PredicateGenerator`